### PR TITLE
AL-8998 Fix resizing for ArrayBuffer in scala-2.12

### PR DESCRIPTION
**Pull request by Mrhs121**

### CODE REVIEW

**Comment by Mrhs121** [comment]
retest this please

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ExpressionSet.scala [hunk 1]
```diff
@@ -58,7 +59,7 @@ object ExpressionSet {
  */
 class ExpressionSet protected(
     private val baseSet: mutable.Set[Expression] = new mutable.HashSet,
-    private val originals: mutable.Buffer[Expression] = new ArrayBuffer)
+    private val originals: mutable.Buffer[Expression] = new BigArrayBuffer)
```
Comment (yugan95) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ExpressionSet.scala:62]: `-- =` 最终会走到 `BigArrayBuffer#remove` 方法，请仔细确认一下这个方法是否需要做调整，对比 scala 社区的实现

**Review by yugan95 — CHANGES_REQUESTED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ExpressionSet.scala (hunk 1 above)
Comment (Mrhs121) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ExpressionSet.scala:62]: 2.12 和 2.13 的实现逻辑上没有区别，2.13 在remove这里稍微优化了一点代码效率

**Comment by Mrhs121** [comment]
retest this please

**Comment by Mrhs121** [comment]
retest this please

**Review by yugan95 — DISMISSED** [review]
LGTM

**Comment by Mrhs121** [comment]
retest this please

**Comment by Mrhs121** [comment]
retest this please

**Comment by Mrhs121** [comment]
retest this please

**Comment by Mrhs121** [comment]
retest this please

**Comment by Mrhs121** [comment]
retest this please

**Comment by Mrhs121** [comment]
retest this please

**Comment by Mrhs121** [comment]
retest this please

**Comment by Mrhs121** [comment]
retest this please

**Review by zheniantoushipashi — APPROVED** [review]

**Pull request merged**